cmd/zpool/zpool_main.c

```diff
@@ -2836,12 +3393,263 @@ get_timestamp_arg(char c)
 }
 
 /*
- * zpool iostat [-gLPv] [-T d|u] [pool] ... [interval [count]]
+ * Return stat flags that are supported by all pools.  "data" should be
+ * initialized to all 0xFFs before running.  It will get ANDed down until
+ * only the flags that are supported on all pools remain.
+ */
+static int
+__get_stat_flags(zpool_handle_t *zhp, void *data)
+{
+	uint64_t *mask = data;
+	nvlist_t *config, *nvroot;
+	uint64_t flags;
+
+	config = zpool_get_config(zhp, NULL);
+	verify(nvlist_lookup_nvlist(config, ZPOOL_CONFIG_VDEV_TREE,
+	    &nvroot) == 0);
+
+	/* What stats does our zfs module support? */
+	if (nvlist_lookup_uint64(nvroot, ZPOOL_CONFIG_VDEV_STATS_FLAGS,
+		&flags) != 0) {
+		/* Probably an old version of the zfs module.  Use defaults. */
+		flags = BIT(VS_DEFAULT);
+	}
+
+	*mask = *mask & flags;
+	return (0);
+}
+
+/*
+ * Newer versions of the zfs module report back a bitmask of all the stats
+ * it supports.  Return the bitmask of stats that are supported on all pools.
+ */
+static uint64_t
+get_stat_flags(zpool_list_t *list)
+{
+	uint64_t mask = -1; /* Start off with all 0xFFs */
+
+	/*
+	 * __get_stat_flags() will lop off bits from "mask" until only the
+	 * flags that are supported on all pools remain.
```

Comment:
Currently all pools share the same kernel modules so it's not clear to me exactly how different pools could support different flags.  However, I'm all for making this as generic as possible because I have no doubt that we might eventually find a case for that, so I'm good with this.